test/test_nn.py

```diff
@@ -5961,6 +5961,13 @@ def test_bce_with_logits_raises_if_target_and_input_are_different_size(self):
         with self.assertRaises(ValueError):
             nn.BCEWithLogitsLoss()(input, target)
 
+    def test_bce_with_logits_raises_if_pos_weight_length_and_num_classes_are_different_size(self):
+        target = torch.rand(5, 3)
+        input = torch.rand(5, 3)
+        pos_weight = torch.tensor([0.1, 0.9])
+        with self.assertRaises(ValueError):
```

Comment:
We should use `self.assertRaisesRegex` to check that the error message matches what we expect it should be.